app/src/main/java/org/thoughtcrime/securesms/conversation/v2/input_bar/InputBar.kt

```diff
@@ -55,6 +66,12 @@ class InputBar : RelativeLayout, InputBarEditTextDelegate, QuoteViewDelegate, Li
         get() { return binding.inputBarEditText.text?.toString() ?: "" }
         set(value) { binding.inputBarEditText.setText(value) }
 
+    // Keep track of when the user pressed the record voice message button, the duration that
+    // they held record, and the current audio recording mechanism state.
+    var voiceMessageStartMS    = 0L
+    var voiceMessageDurationMS = 0L
+    var voiceRecorderState = VoiceRecorderState.Idle
```

Comment:
Nit: can these be `private`?

Reply:
There some external writes to `voiceMessageDurationMS` and `voiceRecorderState` - best we can do is probably something like this?

```
 private var voiceMessageStartMS = 0L
    var voiceMessageDurationMS = 0L
        private set
    internal var voiceRecorderState = VoiceRecorderState.Idle

    fun setVoiceMessageDuration(durationMS: Long)        { voiceMessageDurationMS = durationMS }
    fun setVoiceRecorderState(state: VoiceRecorderState) { voiceRecorderState = state          }
```

Can make the change if you'd prefer to use the above.

Reply:
Ah if it needs public access then I'd prefer to leave it as is.

Reply:
Reverted